src/main/java/com/mcmoddev/lib/API/IMaterialApi.java

```diff
@@ -0,0 +1,37 @@
+package com.mcmoddev.lib.API;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+import com.mcmoddev.lib.material.MMDMaterial;
+
+import net.minecraft.block.Block;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemBlock;
+import net.minecraftforge.fluids.BlockFluidBase;
+import net.minecraftforge.fluids.Fluid;
+
+public interface IMaterialApi extends IMMDApi {
+	default String getApiName() {
+		return "MMD Material Creation Api";
+	}
+	MMDMaterial createMaterial( String name, int hardness, int durability, int magicAffinity, int color, boolean glows );

```

Comment:
I think a builder pattern would make this a lot better. 4 consecutive integers and a boolean would be difficult to read.

Reply:
At the moment this matches what currently exists, actually, though I think we use floats... But yeah, a builder might be a better pattern here. Added to the notes.